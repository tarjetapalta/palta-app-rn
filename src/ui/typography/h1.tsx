import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const H1: FunctionComponent<TextProps> = ({
  /** Test id for test */
  testID = 'H1',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xxl"
    weight="bold"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
