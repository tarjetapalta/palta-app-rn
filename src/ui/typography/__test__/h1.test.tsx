import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { H1 } from '../h1';

describe('h1 component', () => {
  it('basic H1 with color "text", emphasis "high" and testID "H1"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H1>Some text</H1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic H1 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H1 color="primary" emphasis="medium">
            Some text
          </H1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <H1 testID="custom testID">Some text</H1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
