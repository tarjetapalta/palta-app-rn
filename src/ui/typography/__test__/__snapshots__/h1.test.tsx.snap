// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1 component basic H1 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="xxl"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 28,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="H1"
  weight="bold"
>
  Some text
</Text>
`;

exports[`h1 component basic H1 with color "text", emphasis "high" and testID "H1" 1`] = `
<Text
  numberOfLines={0}
  size="xxl"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 28,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="H1"
  weight="bold"
>
  Some text
</Text>
`;

exports[`h1 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="xxl"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Bold",
        "fontSize": 28,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="bold"
>
  Some text
</Text>
`;
